app/models/widget.rb

```diff
@@ -1,4 +1,7 @@
 class Widget < ApplicationRecord
   validates :name, presence: true, uniqueness: true
   validates :order, presence: true, uniqueness: true
+
+  scope :all_in_order, -> { order(order: :asc) }
+  scope :enabled, -> { where(enabled: true).order(order: :asc) }
```

Comment:
I don't think it makes sense to chain order into the enabled scope since we'd want it to be explicit. May instead want to use `default_scope` to define the sort order for all queries.

Reply:
@cbarber can you elucidate more what you mean here? I'm not following entirely.

Reply:
There's two parts:
* scopes should be very explicit since they are chainable e.g. you wouldn't write `scope :enabled, -> { where(enabled: true).order(order: :asc) }`. Instead you would write `scope :enabled, -> { where(enabled: true) }` and use `Widget.enabled.all_in_order.all`.
* we likely always want our widgets to be selected in order. in that case, we can omit defining `scope :all_in_order ...` and simply define `default_scope => { order(order: :asc) }`. Now *every* query for widgets will be ordered.